bilge-impl/src/from_bits.rs

```diff
@@ -40,28 +40,36 @@ fn analyze_enum(variants: Iter<Variant>, name: &Ident, internal_bitsize: BitSize
         abort_call_site!("enum fills its bitsize but has fallback variant"; help = "remove `#[fallback]` from this enum")
     }
 
-    let mut value_assigner = EnumVariantValueAssigner::new(internal_bitsize);
+    let mut assigner = DiscriminantAssigner::new(internal_bitsize);
 
     variants.map(|variant| {
         let variant_name = &variant.ident;
-        let variant_value = value_assigner.assign(variant);
-
-        // might be useful for not generating "1u128 -> Self::Variant"
-        let variant_value: Expr = syn::parse_str(&variant_value.to_string()).unwrap_or_else(unreachable);
-
-        let from_int_match_arm = quote! {
-            #variant_value => Self::#variant_name,
+        let variant_value = assigner.assign_unsuffixed(variant);
+
+        let from_int_match_arm = match fallback {
+            Some(fallback) if fallback.is_fallback_variant(variant_name) => {
+                // this value will be handled by the catch-all arm
+                quote!()
+            },
+            _ => quote! { 
+                #variant_value => Self::#variant_name, 
+            },
         };
-
-        let to_int_match_arm = quote! {
-            #name::#variant_name => Self::new(#variant_value),
+        
+        let to_int_match_arm = match fallback {
+            Some(fallback) if fallback.is_with_value() && fallback.is_fallback_variant(variant_name) => quote! { 
+                #name::#variant_name(number) => number, 
```

Comment:
correct!

Reply:
unfortunately just this change would not be enough here, since our `from_int` arms have no catch-all arm. so we'll get a missing match arm error for the fallback variant.

Reply:
oh right, then let's leave it like that